### PR TITLE
Docs: Add a note about rollover issues

### PR DESCRIPTION
**Pull request by nik9000**
6.4 has a bad bug where it won't if any of the shards on the node have
been rolled over. This documents that in the known issues for 6.4.0 and
links to the fix in the bug fixes section of 6.4.1.

Relates to #33394

### CODE REVIEW

File: docs/reference/release-notes/6.4.asciidoc [hunk 1]
```diff
@@ -55,8 +55,12 @@ coming[6.4.1]
...
 //=== Enhancements
 
-//[float]
-//=== Bug Fixes
+[float]
+=== Bug Fixes
+
+Rollover::
+{es} can once again start if any shards on the node have been
+<<indices-rollover-index, rolled over>>. {pull}33394[#33394]
```
Comment (jasontedor) [inline docs/reference/release-notes/6.4.asciidoc]: I think this needs to wait to be added until 6.4.1 release time?

File: docs/reference/release-notes/6.4.asciidoc [hunk 2]
```diff
@@ -70,6 +74,11 @@ coming[6.4.1]
 [float]
 === Known issues
 
+Rollover::
+{es} can't start if any of the shards on the node have been
+<<indices-rollover-index, rolled over>>. (issue: {issue}33316[#33316])
```
Comment (jasontedor) [inline docs/reference/release-notes/6.4.asciidoc]: I think that we need to highlight this more loudly (and the existing known issue for 6.4.0. Please see #32135 for how I think this should be done.

File: docs/reference/release-notes/6.4.asciidoc [hunk 3]
```diff
@@ -170,15 +177,15 @@ Java High Level REST Client::
 * Rest High Level client: Add List Tasks {pull}29546[#29546] (issue: {issue}27205[#27205])
 
 Machine learning::
-* Detectors now support {stack-ov}/ml-rules.html[custom rules] that enable the 
-user to improve machine learning results by providing some domain-specific 
+* Detectors now support {stack-ov}/ml-rules.html[custom rules] that enable the
```
Comment (jasontedor) [inline docs/reference/release-notes/6.4.asciidoc]: Would you avoid the formatting changes?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left some comments. I think we should put this on the 6.4.0 blog too.

File: docs/reference/release-notes/6.4.asciidoc (hunk 1 above)
Comment (nik9000) [reply docs/reference/release-notes/6.4.asciidoc]: We're already generated 6.4.1 release notes with a "coming in 6.4.1" note. I think it is ok to add this to that page.

File: docs/reference/release-notes/6.4.asciidoc (hunk 3 above)
Comment (nik9000) [reply docs/reference/release-notes/6.4.asciidoc]: Sure.

**Comment by nik9000** [comment]
@jasontedor I've pushed an update. Could you have another look?

**Review by lcawl — APPROVED** [review]
I haven't tried building it, but it LGTM

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by nik9000** [comment]
Thanks for all of your suggestions @jasontedor and @lcawl!

**Pull request merged**